providers/amazon/src/airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -931,7 +932,51 @@ def get_file_metadata(
         max_items: int | None = None,
     ) -> list:
         """
-        List metadata objects in a bucket under prefix.
+        [DEPRECATED] Retrieve metadata objects from a bucket under a prefix.
+
```

Comment:
this is optional. Can we add 
`.. deprecated:: <version>` 

Above will show up in docs. similar to [this](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/_api/airflow/providers/amazon/aws/hooks/glue/index.html#airflow.providers.amazon.aws.hooks.glue.GlueJobHook.list_jobs)